Automattic/pocket-casts-android
# Add basic navigation for new onboarding flow

| 📘 Project: #524  |
| --- |

## Description
This PR adds the basic high-level framework for the new onboarding flow.

## Testing Instructions

1. Open the app
2. ✅ Verify that the onboarding flow does NOT open
3. Update the `ONBOARDING_ENABLED` build flag in `base.gradle` to be "true"
4. ✅ Verify that the onboarding flow opens
5. Admire the amazing UI that has been implemented
6. ✅ Verify that tapping "Not Now" exits the onboarding flow and takes you to the app
7. For each of the other buttons ("Continue with Google", "Sign Up Free", and "Log in"):
    1. ✅ Verify that tapping the button opens a new screen
    2. Again, admire the amazing UI of this new screen
    3. ✅ Verify that swiping back (or tapping the back arrow if there is one) returns to the top-level screen and that swiping back a second time exits the app.

## Checklist
- [ ] Any jetpack compose components I added or changed are covered by compose previews (will handle this when implementing the screens)
 
#### I have tested any UI changes...
